api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -303,4 +303,68 @@ default boolean dropTable(TableIdentifier identifier) {
    * @throws NoSuchTableException if the table does not exist
    */
   Table loadTable(TableIdentifier identifier);
+
+  /**
+   * Create a namespace in the catalog.
+   *
+   * @param namespace {@link Namespace}.
+   */
+  default void createNamespace(Namespace namespace, Map<String, String> meta) {
```

Comment:
I think that these methods should be in an optional interface that can be implemented, like Spark's `SupportsNamespaces`. Then it isn't necessary to throw `UnsupportedOperationException`.